lib/core/network.dart

```diff
@@ -22,21 +35,45 @@ class Network {
         },
       ),
     );
-    _dio.interceptors.add(RequestIdInterceptor());
+    _enteDio = Dio(
+      BaseOptions(
+        baseUrl: apiEndpoint,
+        connectTimeout: kConnectTimeout,
+        headers: {
+          HttpHeaders.userAgentHeader: FkUserAgent.userAgent,
+          'X-Client-Version': packageInfo.version,
+          'X-Client-Package': packageInfo.packageName,
+        },
+      ),
+    );
+    enteDio.interceptors.add(EnteRequestInterceptor(preferences));
   }
 
   Network._privateConstructor();
 
   static Network instance = Network._privateConstructor();
 
   Dio getDio() => _dio;
+
+  Dio get enteDio => _enteDio;
 }
 
-class RequestIdInterceptor extends Interceptor {
+class EnteRequestInterceptor extends Interceptor {
+  final SharedPreferences preferences;
+
+  EnteRequestInterceptor(this.preferences);
+
   @override
   void onRequest(RequestOptions options, RequestInterceptorHandler handler) {
+    if (kDebugMode) {
+      assert(options.baseUrl == Network.apiEndpoint, "request should be");
+    }
     // ignore: prefer_const_constructors
     options.headers.putIfAbsent("x-request-id", () => Uuid().v4().toString());
+    final String? tokenValue = preferences.getString(Configuration.tokenKey);
```

Comment:
Why not simply read from `Configuration` directly each time? We are using `Configuration.tokenKey` anyway

Reply:
I didn't wanted to change the init order. Eventually I want to kill Configuration and have separate stores.
Network init happens before Config init.